### PR TITLE
[IDP-782] Add Error description in error message

### PR DESCRIPTION
**Pull request by PrincessMadMath**
- Only build string when needed

### CODE REVIEW

**Review by starcraft66 — APPROVED** [review]

File: src/Workleap.Extensions.Http.Authentication.ClientCredentialsGrant/ClientCredentialsTokenEndpointService.cs [hunk 1]
```diff
@@ -71,32 +72,52 @@ public async Task<ClientCredentialsToken> RequestTokenAsync(string clientName, C
...
         }
 
         if (string.IsNullOrEmpty(response.AccessToken))
         {
-            throw new ClientCredentialsException(exceptionMessagePrefix + ": result was empty");
+            throw new ClientCredentialsException(GetErrorMessage(clientName, response, "result was empty"));
+        }
+    }
+
+    private static string GetErrorMessage(string clientName, TokenResponse response, params string[] additionalMessageParts)
```
Comment (asimmon) [inline src/Workleap.Extensions.Http.Authentication.ClientCredentialsGrant/ClientCredentialsTokenEndpointService.cs:102]: Why using a params string array here? We only provide zero or one item

File: src/Workleap.Extensions.Http.Authentication.ClientCredentialsGrant/ClientCredentialsTokenEndpointService.cs [hunk 2]
```diff
@@ -71,32 +72,52 @@ public async Task<ClientCredentialsToken> RequestTokenAsync(string clientName, C
...
+        {
+            exceptionMessagePrefixBuilder.Append($": {response.ErrorDescription}");
+        }
+
+        foreach (var additionalMessagePart in additionalMessageParts)
+        {
+            exceptionMessagePrefixBuilder.Append($": {additionalMessagePart}");
         }
+        
+        return exceptionMessagePrefixBuilder.ToString();
```
Comment (asimmon) [inline src/Workleap.Extensions.Http.Authentication.ClientCredentialsGrant/ClientCredentialsTokenEndpointService.cs:121]: With all these semicolon, how the final message might look like?

> An error occured while retrieving token for client '{clientName}': something: foobar: result was empty

File: src/Workleap.Extensions.Http.Authentication.ClientCredentialsGrant/ClientCredentialsTokenEndpointService.cs [hunk 3]
```diff
@@ -71,32 +72,52 @@ public async Task<ClientCredentialsToken> RequestTokenAsync(string clientName, C
...
         if (string.IsNullOrEmpty(response.AccessToken))
         {
-            throw new ClientCredentialsException(exceptionMessagePrefix + ": result was empty");
+            throw new ClientCredentialsException(GetErrorMessage(clientName, response, "result was empty"));
+        }
+    }
+
+    private static string GetErrorMessage(string clientName, TokenResponse response, params string[] additionalMessageParts)
+    {
+        var exceptionMessagePrefixBuilder = new StringBuilder($"An error occured while retrieving token for client '{clientName}");
```
Comment (asimmon) [inline src/Workleap.Extensions.Http.Authentication.ClientCredentialsGrant/ClientCredentialsTokenEndpointService.cs:104]:
```suggestion
        var exceptionMessagePrefixBuilder = new StringBuilder($"An error occurred while retrieving token for client '{clientName}'");
```

File: src/Workleap.Extensions.Http.Authentication.ClientCredentialsGrant/ClientCredentialsTokenEndpointService.cs (hunk 2 above)
Comment (asimmon) [reply src/Workleap.Extensions.Http.Authentication.ClientCredentialsGrant/ClientCredentialsTokenEndpointService.cs:121]: Or maybe these details are usually exclusive 🤔 I don't remember

Comment (asimmon) [reply src/Workleap.Extensions.Http.Authentication.ClientCredentialsGrant/ClientCredentialsTokenEndpointService.cs:121]: Nah let's keep it that way, it's good enough

**Review by asimmon — APPROVED** [review]

**Pull request merged**